app/ts/McdaApp/Workspace/CurrentTab/Preferences/EquivalentChange/EquivalentChangeStatement/EquivalentChangeInput/EquivalentChangeValueInput.tsx

```diff
@@ -0,0 +1,102 @@
+import {Grid, InputAdornment, Popover, TextField} from '@material-ui/core';
+import {CurrentSubproblemContext} from 'app/ts/McdaApp/Workspace/CurrentSubproblemContext/CurrentSubproblemContext';
+import {SettingsContext} from 'app/ts/McdaApp/Workspace/SettingsContext/SettingsContext';
+import {getUnitLabel} from 'app/ts/util/getUnitLabel';
+import React, {ChangeEvent, useContext, useEffect, useState} from 'react';
+import {EquivalentChangeContext} from '../../EquivalentChangeContext/EquivalentChangeContext';
+import {getTheoreticalRange} from '../../equivalentChangeUtil';
+
+export default function EquivalentChangeValueInput({
+  anchorElement,
+  isDialogOpen,
+  closeDialog
+}: {
+  anchorElement: HTMLButtonElement;
+  closeDialog: () => void;
+  isDialogOpen: boolean;
+}): JSX.Element {
+  const {referenceValueBy, referenceCriterion, setReferenceValueBy} =
+    useContext(EquivalentChangeContext);
+  const {stepSizesByCriterion} = useContext(CurrentSubproblemContext);
+  const {getUsePercentage} = useContext(SettingsContext);
+  const [inputError, setInputError] = useState<string>('');
+
+  const unit = referenceCriterion.dataSources[0].unitOfMeasurement;
```

Comment:
I've seen this `criterion.dataSources[0].unitOfMeasurement` bit so often by now I think it might be worth extracting to a function in case we ever want to do it differently